storm-core/src/clj/org/apache/storm/stats.clj

```diff
@@ -188,14 +168,12 @@
 
 (defn value-bolt-stats!
   [^BoltExecutorStats stats]
-  (cleanup-bolt-stats! stats)
   (merge (value-common-stats (:common stats))
          (value-stats stats BOLT-FIELDS)
          {:type :bolt}))
 
 (defn value-spout-stats!
   [^SpoutExecutorStats stats]
-  (cleanup-spout-stats! stats)
```

Comment:
Thanks @revans2 . Cancelling the task during the executor shutdown makes sense. I will modify the PR accordingly and restore the close functions.